generator/internal/descriptor_utils.cc

```diff
@@ -634,20 +634,22 @@ bool CheckParameterCommentSubstitutions() {
 
 /// If a service name mapping exists, return the new name.
 /// Parses a command line argument in the form:
-/// {"service_name_mappings": "service_a:new_service_a,service:new_service"}.
+/// {"service_name_mappings": "service_a=new_service_a,service=new_service"}.
 std::string GetEffectiveServiceName(VarsDictionary const& vars,
-                                    std::string const& name) {
+                                    absl::string_view name) {
   auto service_name_mappings = vars.find("service_name_mappings");
-  if (service_name_mappings == vars.end()) return name;
-
-  std::vector<std::string> mappings =
-      absl::StrSplit(service_name_mappings->second, ',');
-  for (auto& mapping : mappings) {
-    std::vector<std::string> kv = absl::StrSplit(mapping, ':');
-    if (kv.size() != 2) continue;
-    if (kv[0] == name) return kv[1];
+  if (service_name_mappings != vars.end()) {
+    std::map<absl::string_view, absl::string_view> map;
+    for (absl::string_view arg :
+         absl::StrSplit(service_name_mappings->second, ',')) {
+      map.insert(absl::StrSplit(arg, absl::MaxSplits('=', 1)));
+    }
+    if (map.find(name) != map.end()) {
+      return std::string(map.find(name)->second.data(),
+                         map.find(name)->second.size());
+    }
```

Comment:
Saving the result of the first `find()` would be preferable to running the same lookup three times.

Reply:
Done.